test/pthread/test_pthread_c_thread_local.c

```diff
@@ -0,0 +1,35 @@
+// Copyright 2025 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <pthread.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <stdio.h>
+
+static _Thread_local int thread_local = 0;
```

Comment:
Would it perhaps make sense to do ` = 3` here and assert that `thread_local` starts with this value in moth `main` and in `thread_start?